packages/next/build/analysis/get-page-static-info.ts

```diff
@@ -239,6 +240,23 @@ function warnAboutUnsupportedValue(
   warnedUnsupportedValueMap.set(pageFilePath, true)
 }
 
+function getUserConfig(
+  config: any,
+  middlewareConfig: Partial<MiddlewareConfig>
+) {
+  if (!config) return
+
+  const userConfig = config && { ...config }
+  if (userConfig) {
+    delete userConfig.runtime
+    if (middlewareConfig.matchers) {
+      delete userConfig.matcher
```

Comment:
suggestion: how about throwing an error like `you can't specify both "matcher" and "matchers"` instead of discarding `matcher` here silently?

Reply:
This is not a validation, I just don’t want to serialize the matcher again (it already exists in the middleware manifest)